elasticsearch/shield.sls

```diff
@@ -78,45 +78,120 @@ copy_shield_config:
     - require_in:
       - service: start_elasticsearch
 
-/etc/elasticsearch/server.crt:
+/etc/elasticsearch/ca:
+  file:
+    - recurse
+    - source: salt://elasticsearch/ca
+    - template: jinja
+    - user: root
+    - group: root
+    - file_mode: 644
+    - require:
+      - pkg: elasticsearch
+
+/etc/elasticsearch/ca/private/cakey.pem:
   file:
     - managed
     - user: root
     - group: root
-    - mode: 664
-    - contents_pillar: elasticsearch:encryption:certificate
+    - mode: 400
+    - makedirs: true
+    - contents_pillar: elasticsearch:encryption:ca_key
+    - require:
+      - file: /etc/elasticsearch/ca
 
-/etc/elasticsearch/server.key:
+/etc/elasticsearch/ca/certs/cacert.pem:
   file:
     - managed
     - user: root
     - group: root
-    - mode: 664
-    - contents_pillar: elasticsearch:encryption:private_key
+    - mode: 400
+    - makedirs: true
+    - contents_pillar: elasticsearch:encryption:ca_cert
+    - require:
+      - file: /etc/elasticsearch/ca
+
+# Delete before re-creating to ensure idempotency
+delete-truststore:
```

Comment:
hmm, I'm not entirely sure, I want to say ES loads it on start and would require a restart to pick up changes to the file, so shouldn't be an issue if it's gone for a split second